% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\usepackage[utf8]{inputenc}
\DeclareUnicodeCharacter{00A0}{\nobreakspace}
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}
\usepackage{multirow}


\title{TAPIS Documentation}
\date{November 19, 2015}
\release{1.1.2}
\author{Mike Hamilton}
\newcommand{\sphinxlogo}{}
\renewcommand{\releasename}{Release}
\makeindex

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\expandafter\def\csname PYG@tok@gd\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gu\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@gt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\expandafter\def\csname PYG@tok@gs\endcsname{\let\PYG@bf=\textbf}
\expandafter\def\csname PYG@tok@gr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@cm\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@vg\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@m\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@mh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@cs\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\colorbox[rgb]{1.00,0.94,0.94}{\strut ##1}}}
\expandafter\def\csname PYG@tok@ge\endcsname{\let\PYG@it=\textit}
\expandafter\def\csname PYG@tok@vc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@il\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@go\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.20,0.20,0.20}{##1}}}
\expandafter\def\csname PYG@tok@cp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@gi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gh\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@ni\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\expandafter\def\csname PYG@tok@nl\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\expandafter\def\csname PYG@tok@nn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@no\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\expandafter\def\csname PYG@tok@na\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@nd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\expandafter\def\csname PYG@tok@ne\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\expandafter\def\csname PYG@tok@si\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\expandafter\def\csname PYG@tok@s2\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@vi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@nt\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\expandafter\def\csname PYG@tok@nv\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@s1\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@gp\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@sh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ow\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@sx\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@bp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c1\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@kc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@mf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@err\endcsname{\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}
\expandafter\def\csname PYG@tok@mb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@ss\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\expandafter\def\csname PYG@tok@sr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\expandafter\def\csname PYG@tok@mo\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@mi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@o\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PYG@tok@kr\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@s\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@kp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@w\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\expandafter\def\csname PYG@tok@kt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\expandafter\def\csname PYG@tok@sc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@k\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@se\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sd\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}

\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZam{\char`\&}
\def\PYGZlt{\char`\<}
\def\PYGZgt{\char`\>}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZhy{\char`\-}
\def\PYGZsq{\char`\'}
\def\PYGZdq{\char`\"}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\renewcommand\PYGZsq{\textquotesingle}

\begin{document}

\maketitle
\tableofcontents
\phantomsection\label{index::doc}


Contents:


\chapter{Get TAPIS}
\label{installation:welcome-to-tapis-s-documentation}\label{installation::doc}\label{installation:get-tapis}

\section{Required packages}
\label{installation:required-packages}
The following packages are required for TAPIS core functions.
Version numbers correspond to those tested during development.
\begin{itemize}
\item {} 
\href{http://splicegrapher.sourceforge.net/}{SpliceGrapher} (v0.2.4 )

\item {} 
\href{https://code.google.com/p/pysam/}{Pysam} (v0.8.1)

\item {} 
\href{http://matplotlib.org/}{matplotlib} (v1.3.1)

\item {} 
\href{https://pypi.python.org/pypi/bx-python/0.7.3}{bx-python} (v0.5.0)

\item {} 
\href{http://www.numpy.org/}{NumPy} (v1.8.2)

\item {} 
\href{http://research-pub.gene.com/gmap/}{GMAP} (v2015-07-23)

\end{itemize}


\section{Download}
\label{installation:download}
current release: v1.1.2

TAPIS is hosted on bitbucket \href{https://bitbucket.org/comp\_bio/tapis}{https://bitbucket.org/comp\_bio/tapis}


\section{Install}
\label{installation:install}
\begin{Verbatim}[commandchars=\\\{\}]
\PYGZdl{} tar zxvf tapis\PYGZus{}\PYGZlt{}version\PYGZgt{}.tgz
\PYGZdl{} cd tapis\PYGZus{}\PYGZlt{}version\PYGZgt{}.tgz
\PYGZdl{} python setup.py install
\end{Verbatim}

\begin{notice}{note}{Note:}
To install in a user directory, use the option:

\begin{Verbatim}[commandchars=\\\{\}]
\PYGZhy{}\PYGZhy{}home=/Path/To/Local/Library
\end{Verbatim}
\end{notice}


\chapter{Tutorial}
\label{tutorial::doc}\label{tutorial:tutorial}\label{tutorial:gmap}
This tutorial is meant as a complete walk-through for identifying
transcripts and poly(A) sites from PacBio reads.
\begin{enumerate}
\item {} 
Align and clean reads

\item {} 
Cluster reads and analyze transcripts and poly(A) sites

\end{enumerate}


\section{Align reads}
\label{tutorial:align-reads}
\textbf{TAPIS} accepts any sorted, indexed BAM file for
long reads but it provides a method that cleans and aligns
reads with high accuracy and efficiency. To align and clean reads
use the following provided script \textbf{alignPacBio.py}.  Before
running the script, you will need to run \textbf{gmap\_build} to
make a genome reference index.

\begin{Verbatim}[commandchars=\\\{\}]
usage: alignPacBio.py [\PYGZhy{}h] [\PYGZhy{}v] [\PYGZhy{}i ITERATIONS] [\PYGZhy{}e EDR] [\PYGZhy{}o OUTDIR]
                      [\PYGZhy{}p PROCS] [\PYGZhy{}K MAXINTRON]
                      indexesDir indexName reference fasta

Iteratively fix aligned reads using reference genome

positional arguments:
  indexesDir            directory to gmap indexes
  indexName             name of gmap index
  reference             Reference sequence
  fasta                 Reads to align

optional arguments:
  \PYGZhy{}h, \PYGZhy{}\PYGZhy{}help            show this help message and exit
  \PYGZhy{}v, \PYGZhy{}\PYGZhy{}verbose         Verbose mode
  \PYGZhy{}i ITERATIONS, \PYGZhy{}\PYGZhy{}iterations ITERATIONS
                        Number of aligment iterations, default=3
  \PYGZhy{}e EDR, \PYGZhy{}\PYGZhy{}edr EDR     Edit distance ratio, default=10
  \PYGZhy{}o OUTDIR, \PYGZhy{}\PYGZhy{}outdir OUTDIR
                        Output directory, default=./cleanedAlignments
  \PYGZhy{}p PROCS, \PYGZhy{}\PYGZhy{}procs PROCS
                        Number of processors, default=1
  \PYGZhy{}K MAXINTRON, \PYGZhy{}\PYGZhy{}maxIntron MAXINTRON
                        maximum intron length for gmap, default=8000
\end{Verbatim}


\section{Creating indexed, sorted BAM files}
\label{tutorial:creating-indexed-sorted-bam-files}
If your cleaned/aligned reads are in the form of a SAM file, you can convert it to a indexed,
sorted BAM file using \textbf{convertSam.py}.

\begin{Verbatim}[commandchars=\\\{\}]
usage: convertSam.py [\PYGZhy{}h] [\PYGZhy{}o BAMFILE] [\PYGZhy{}p PROCS] [\PYGZhy{}m MEMORY] [\PYGZhy{}v] samfile

Generate sorted BAM and index files for given SAM file

positional arguments:
  samfile               Samfile to convert

optional arguments:
  \PYGZhy{}h, \PYGZhy{}\PYGZhy{}help            show this help message and exit
  \PYGZhy{}o BAMFILE, \PYGZhy{}\PYGZhy{}outfile BAMFILE
                        Name of converted BAM file [default=\PYGZlt{}sambase\PYGZgt{}.bam]
  \PYGZhy{}p PROCS, \PYGZhy{}\PYGZhy{}procs PROCS
                        Number of processors to use for BAM sorting (default
                        1)
  \PYGZhy{}m MEMORY, \PYGZhy{}\PYGZhy{}memory MEMORY
                        Max memory (in GBs) for each processor used for BAM
                        sorting (default 2)
  \PYGZhy{}v, \PYGZhy{}\PYGZhy{}verbose         Print verbose output
\end{Verbatim}


\section{Running \textbf{TAPIS}}
\label{tutorial:running-tapis}
\begin{Verbatim}[commandchars=\\\{\}]
\PYGZdl{} run\PYGZus{}tapis.py \PYGZhy{}\PYGZhy{}help
usage: run\PYGZus{}tapis.py [\PYGZhy{}h] [\PYGZhy{}v] [\PYGZhy{}p] [\PYGZhy{}o OUTDIR] [\PYGZhy{}t TRIMMAX] [\PYGZhy{}w W]
                    [\PYGZhy{}m MINDIST]
                    geneModel bamfile

Assemble transcripts from PacBio alignments

positional arguments:
  geneModel             Gene models annotation file (GFF/GTF)
  bamfile               Aligned reads file (sorted and indexed)

optional arguments:
  \PYGZhy{}h, \PYGZhy{}\PYGZhy{}help            show this help message and exit
  \PYGZhy{}v, \PYGZhy{}\PYGZhy{}verbose         Verbose mode
  \PYGZhy{}p, \PYGZhy{}\PYGZhy{}plot            Plot novel gene graphs and poly(A) figures, default is
                        no plotting
  \PYGZhy{}o OUTDIR, \PYGZhy{}\PYGZhy{}outdir OUTDIR
                        Output directory for TAPIS results, default=tapis\PYGZus{}out
  \PYGZhy{}t TRIMMAX, \PYGZhy{}\PYGZhy{}trimMax TRIMMAX
                        Maximum length of read trimming to tolerate, default=5
  \PYGZhy{}w W, \PYGZhy{}\PYGZhy{}w W           Width of peaks when searching for poly(A) sites,
                        default=5
  \PYGZhy{}m MINDIST, \PYGZhy{}\PYGZhy{}minDist MINDIST
                        Minimum distance between any two poly(A) sites,
                        default=20
\end{Verbatim}

While \textbf{TAPIS} offers many options, default values should work
for most cases.


\section{Interpreting \textbf{TAPIS} output}
\label{tutorial:interpreting-tapis-output}
\textbf{TAPIS} builds an output directory as follows:

\begin{Verbatim}[commandchars=\\\{\}]
\PYGZdl{} tree my\PYGZus{}result
  tapis\PYGZus{}out
  \textbar{}\PYGZhy{}\PYGZhy{} polyAFigures
  \textbar{}   \textbar{}\PYGZhy{}\PYGZhy{} gene1.png
  \textbar{}   \textbar{}\PYGZhy{}\PYGZhy{} gene2.pbg
  \textbar{}   \textbar{}\PYGZhy{}\PYGZhy{} ...
  \textbar{}   \textbar{}\PYGZhy{}\PYGZhy{} geneN.png
  \textbar{}\PYGZhy{}\PYGZhy{} novelGraphs
  \textbar{}   \textbar{}\PYGZhy{}\PYGZhy{} chrom\PYGZus{}start\PYGZus{}end\PYGZus{}strand.pdf
  \textbar{}   \textbar{}\PYGZhy{}\PYGZhy{} ...
  \textbar{}\PYGZhy{}\PYGZhy{} assembled.gtf
  \textbar{}\PYGZhy{}\PYGZhy{} novelGenes.csv
  \textbar{}\PYGZhy{}\PYGZhy{} novelGenes.fa
  \textbar{}\PYGZhy{}\PYGZhy{} polyA\PYGZus{}summary.csv
\end{Verbatim}
\begin{itemize}
\item {} 
\textbf{polyAFigures} - contains poly(A) site depictions for genes with
at least one poly(A) site supported by long reads.

\item {} 
\textbf{novelGraphs} - contains splice graph figures for transcripts not found in within any annotated gene.

\item {} 
\textbf{assembled.gtf} - gene models for transcripts detected in long reads

\item {} 
\textbf{novelGenes.csv} - tab-delimited file containing summary of novel genes detected

\end{itemize}


\chapter{Contact}
\label{contact:contact}\label{contact::doc}
\textbf{TAPIS} is developed by \href{http://www.cs.colostate.edu/~marcel77}{Mike Hamilton} at Colorado State
University.

Bug reports and feature requests can be submitted
through \href{https://bitbucket.org/comp\_bio/tapis}{bitbucket}.
\begin{itemize}
\item {} 
\emph{search}

\end{itemize}

\begin{thebibliography}{SG}
\bibitem[SG]{SG}{\phantomsection\label{tutorial:sg} 
Rogers, MF, Thomas, J, Reddy, AS, Ben-Hur, A (2012).
SpliceGrapher: detecting patterns of alternative splicing
from RNA-Seq data in the context of gene models and
EST data. \emph{Genome Biol}., 13, 1:R4.
}
\end{thebibliography}



\renewcommand{\indexname}{Index}
\printindex
\end{document}
